ddtrace/internal/datadog/profiling/stack_v2/src/sampler.cpp

```diff
@@ -67,6 +67,11 @@ _stack_v2_atfork_child()
     // so we don't even reveal this function to the user
     _set_pid(getpid());
     ThreadSpanLinks::postfork_child();
+
+    // `thread_info_map_lock` is a global lock held in echion; it needs to be destroyed and re-initialized to ensure
+    // it has a consistent state
+    thread_info_map_lock.~mutex();
+    new (&thread_info_map_lock) std::mutex;
```

Comment:
```suggestion
    // `thread_info_map_lock` and `task_link_map_lock` are global locks held in echion; they need to be destroyed and re-initialized to ensure
    // they have a consistent state
    thread_info_map_lock.~mutex();
    new (&thread_info_map_lock) std::mutex;
    task_link_map_lock.~mutex();
    new (&task_link_map_lock) std::mutex;
```

Used here https://github.com/DataDog/dd-trace-py/blob/e46e3b5785ee26aec4fbf9e7cd2c13c40a435697/ddtrace/internal/datadog/profiling/stack_v2/src/sampler.cpp#L180